app/controllers/reset_passwords_controller.rb

```diff
@@ -0,0 +1,24 @@
+class ResetPasswordsController < ApplicationController
+  def show
+    user = User.find_by(token: params[:id])
+    if user
+      @token = params[:id]
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    token = params[:token]
+    redirect_to root_path and return if token.blank?
+
+    user = User.find_by(token: params[:token])
+    if user
+      user.update!({password: params[:password], token: nil})

```

Comment:
I had just the check at user creation. I had the password validation but only on create.
```validates :password, presence: true, length: {minimum: 5}, on: :create ```. I have changed so that is always validated (makes more sense). I have created a new pull request with the changes.